React/CxxBridge/RCTCxxBridge.mm

```diff
@@ -372,25 +374,42 @@ - (void)start
     }));
   }
 
+  // TODO(OSS Candidate ISS#2710739) runtime initialization sequencing
+  [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(runtimeInitializationDidEnd:) name:(NSString*)RCTRuntimeInitializationEndNotificationName object:nil];
+  
+  prepareBridge = dispatch_group_create();
+  dispatch_group_enter(prepareBridge);
   // Dispatch the instance initialization as soon as the initial module metadata has
   // been collected (see initModules)
-  dispatch_group_enter(prepareBridge);
   [self ensureOnJavaScriptThread:^{
     [weakSelf _initializeBridge:executorFactory];
-    dispatch_group_leave(prepareBridge);
+    dispatch_group_leave(self->prepareBridge);
   }];
+  
+  // TODO(OSS Candidate ISS#2710739): We can call this multiple times but as long as we've set up the runtime once, we should be all set to continue
+  if (runtimeIsInitialized) {
+    [self runtimeInitializationDidEnd:nil];
+  }
+}
 
-  // Load the source asynchronously, then store it for later execution.
-  dispatch_group_enter(prepareBridge);
+#if TARGET_OS_OSX // TODO(OSS Candidate ISS#2710739)
+- (void)runtimeInitializationDidEnd:(NSEvent*)event {
+#else
+- (void)runtimeInitializationDidEnd:(UIEvent*)event {
+#endif // TARGET_OS_OSX
```

Comment:
No need for the macros here, as they'd accept an `NSNotification*`, but you can also just leave off the argument if you don't need it.